### PR TITLE
fix build issue on clang 15

### LINKED ISSUE #50963: Clang 15 Build failed after #49752
Using Clang 15 build latest code with #49752 will failed:
```

[2023-06-13T23:38:02.611Z] /ClickHouse/src/Formats/CapnProtoSerializer.cpp:754:9: error: missing 'typename' prior to dependent type name 'CapnpType::Reader'
[2023-06-13T23:38:02.611Z]         CapnpType::Reader getData(const ColumnPtr & column, size_t row_num)
[2023-06-13T23:38:02.611Z]         ^~~~~~~~~~~~~~~~~
[2023-06-13T23:38:02.611Z]         typename 
[2023-06-13T23:38:02.611Z] /ClickHouse/src/Formats/CapnProtoSerializer.cpp:804:9: error: missing 'typename' prior to dependent type name 'CapnpType::Reader'
[2023-06-13T23:38:02.611Z]         CapnpType::Reader getData(const ColumnPtr & column, size_t row_num)
[2023-06-13T23:38:02.611Z]         ^~~~~~~~~~~~~~~~~
[2023-06-13T23:38:02.611Z]         typename 
```

Clang 16 is ok

### PR DESCRIPTION
**Pull request by baibaichen**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
...

### Documentation entry for user-facing changes
Closes \#50963

It's a minor change and keeps support for clang-15 to ease the transition period

### CODE REVIEW

**Comment by baibaichen** [comment]
@serxa would you please review this？

File: src/Formats/CapnProtoSerializer.cpp [hunk 1]
```diff
@@ -751,7 +751,7 @@ namespace
     private:
         using Reader = typename CapnpType::Reader;
 
-        CapnpType::Reader getData(const ColumnPtr & column, size_t row_num)
+        typename CapnpType::Reader getData(const ColumnPtr & column, size_t row_num)
```
Comment (antonio2368) [inline src/Formats/CapnProtoSerializer.cpp:754]: you should be able to use Reader, defined 2 lines above

File: src/Formats/CapnProtoSerializer.cpp [hunk 2]
```diff
@@ -801,7 +801,7 @@ namespace
...
```
Comment (antonio2368) [inline src/Formats/CapnProtoSerializer.cpp:804]: same comment as above

**Comment by robot-ch-test-poll3** [comment]
This is an automated comment for commit 86694847c66a170781e5a561940c05e0221966d0 with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/50967/86694847c66a170781e5a561940c05e0221966d0/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

**Comment by baibaichen** [comment]
@antonio2368 can you merge this pr, I think it's too simple, no need to run CI.

**Review by antonio2368 — APPROVED** [review]

**Pull request merged**